homeassistant/components/ecovacs_mqtt/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""Ecovacs mqtt component."""
+from __future__ import annotations
+
+import asyncio
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .controller import EcovacsController
+
+PLATFORMS = [
+    Platform.VACUUM,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    controller = EcovacsController(hass, {**entry.data, **entry.options})
+    await controller.initialize()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = controller
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    # Reload entry when its updated.
+    entry.async_on_unload(entry.add_update_listener(async_reload_entry))
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    # This is called when an entry/configured device is to be removed. The class
+    # needs to unload itself, and remove callbacks. See the classes for further
+    # details
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+
+    if unload_ok:
+        await hass.data[DOMAIN][entry.entry_id].teardown()
```

Comment:
Would it maybe be cleaner to do this in an `entry.async_on_unload()`?

Reply:
Nothing, but maybe it looks better code wise, it was also a question, not a remark :)

Reply:
@MartinHjelmare What's your opinion/suggestion on this topic? Should I leave it here or use `entry.async_on_unload`?